Создание проекта

NikitaGoroshko7 opened the pull request:
https://trello.com/c/XTJkOOg1/20-%D1%81%D0%BE%D0%B7%D0%B4%D0%B0%D0%BD%D0%B8%D0%B5-%D0%BF%D1%80%D0%BE%D0%B5%D0%BA%D1%82%D0%B0
Created ProjectController
Added actions for ProjectController
Added Views
Added general css file and added scripts
Added migration and created table CreateProject in the database
Added UploadedFileService and changed in the ProfileController
Added folder for storage ProjectImages

NikitaGoroshko7 committed:
Создание проекта
https://trello.com/c/XTJkOOg1/20-%D1%81%D0%BE%D0%B7%D0%B4%D0%B0%D0%BD%D0%B8%D0%B5-%D0%BF%D1%80%D0%BE%D0%B5%D0%BA%D1%82%D0%B0
Created ProjectController
Added actions for ProjectController
Added Views
Added general css file and added scripts
Added migration and created table CreateProject in the database
Added UploadedFileService and changed in the ProfileController
Added folder for storage ProjectImages

NikitaGoroshko7 committed:
Deleted some styles

NikitaGoroshko7 committed:
Deleted style

GalayM commented on CrowdFundingApplication/Models/CreateProject.cs:
```diff
@@ -0,0 +1,12 @@
+    public string Description { get; set; }
+    public string Category { get; set; }
+    public int Sum { get; set; }
+    public string Date { get; set; }
```
string -> DateTime

NikitaGoroshko7 replied:
Has been changed

GalayM commented on CrowdFundingApplication/Models/CreateProject.cs:
```diff
@@ -0,0 +1,12 @@
+    public string ImageName { get; set; }
+    public string Description { get; set; }
+    public string Category { get; set; }
+    public int Sum { get; set; }
```
int? Not decimal?

NikitaGoroshko7 replied:
Has been changed

GalayM commented on CrowdFundingApplication/Models/CreateProject.cs:
```diff
@@ -0,0 +1,12 @@
+    public string NameOfProject { get; set; }
+    public string ImageName { get; set; }
+    public string Description { get; set; }
+    public string Category { get; set; }
```
string -> enum

NikitaGoroshko7 replied:
Has been changed

GalayM commented on CrowdFundingApplication/Models/CreateProjectViewModel.cs:
```diff
@@ -0,0 +1,62 @@
+﻿namespace CrowdFundingApplication.Models;
+
+public class CreateProjectViewModel
```
It is View or Edit model? If View then there should be no validation.

NikitaGoroshko7 replied:
It is edit model

GalayM commented on CrowdFundingApplication/Models/CreateProjectViewModel.cs:
```diff
@@ -0,0 +1,62 @@
+    public string Date { get; set; }
+}
+
+public enum Category
```
Each entity in a separate file

NikitaGoroshko7 replied:
Has been changed

GalayM commented on CrowdFundingApplication/Views/Project/Index.cshtml:
```diff
@@ -0,0 +1,80 @@
+}
+
+<form method="post" asp-action="Index">
+<br />
```
Remove br

NikitaGoroshko7 replied:
Has been changed

GalayM commented on CrowdFundingApplication/Views/Project/Index.cshtml:
```diff
@@ -0,0 +1,80 @@
+                                
+                                <div class="col-lg-3 mt-4 mb-5">
+                                    <button class="sub_button btn btn-primary btn-block" disabled id="button" type="submit">Создать проект</button>
+                                    <br />
```
Remove all BRs

NikitaGoroshko7 replied:
Has been changed

GalayM commented on CrowdFundingApplication/Controllers/ProjectController.cs:
```diff
@@ -0,0 +1,74 @@
+
+                var obj = GetCreateProject(model);
+                _db.Add(obj);
+                _db.SaveChanges();
```
It is better to use asynchronous implementation.

NikitaGoroshko7 replied:
Has been changed

NikitaGoroshko7 committed:
Added changes
Deleted br from Views
Added changes in ProjectController
Category has been transferredto another file

NikitaGoroshko7 committed:
Has been deleted the previous migration
Has been added a new migration

NikitaGoroshko7 committed:
Changed name ViewModel on EditProjectViewModel

GalayM commented on CrowdFundingApplication/Models/CreateProjectViewModel.cs:
```diff
@@ -0,0 +1,22 @@
+    public int Sum { get; set; }
+
+    [Required(ErrorMessage = "Не указана дата окончания проекта!")]
+    public string Date { get; set; }
```
string -> DateTime
int -> decimal

GalayM commented on CrowdFundingApplication/Controllers/ProjectController.cs:
```diff
@@ -0,0 +1,74 @@
+    {
+        var project = new CreateProject
+        {
+            Id = Guid.NewGuid(),
```
You do not need to fill in the ID, they must be generated by themselves

GalayM commented on CrowdFundingApplication/Models/Category.cs:
```diff
@@ -0,0 +1,41 @@
+﻿namespace CrowdFundingApplication.Models;
+
+public enum Category
```
Category -> CategoryEnum

GalayM commented on CrowdFundingApplication/wwwroot/js/site.js:
```diff
@@ -1,4 +1,27 @@
+}
+
+function showCount2() {
+    resultid.innerHTML = 125;
```
move 125 to viable

NikitaGoroshko7 committed:
Added changes
